symai/components.py

```diff
@@ -301,3 +301,40 @@ def _func(_):
         obj._sym_return_type = _type
         return obj(*args, **kwargs)
 
+
+class SimilarityClassification(Expression):
+    def __init__(self, classes: List[str], metric: str = 'cosine'):
+        super().__init__()
+        self.classes = classes
+        self.metric  = metric
+
+    def embed_classes(self):
```

Comment:
Should this not use the @cache decorator?

Reply:
I forgot to tell you in our call. It's getting nasty here. Because cache is a decorator itself, you can't parameterize it from within the class, i.e. change its `in_memory=True/False`. We can't set a default `True` value in here without first telling the user he needs to always be aware of the .cache. I can set its default to `True` for now until we find a solution.

Reply:
In this case I would parametrize the class constructor with the in_memory property and set it to False to make the assure a user desired behavior and not a nasty side effect